src/classes/Recipe.js

```diff
@@ -0,0 +1,39 @@
+class Recipe {
+    constructor(recipeDetails) {
+        this.id = recipeDetails.id || this.generateRandomId();
+        this.name = recipeDetails.name;
+        this.image = recipeDetails.image;
+        this.portions = recipeDetails.portions || [];
+        this.instructions = recipeDetails.instructions || [];
+        this.tags = recipeDetails.tags || [];
+
+    }
+    generateRandomId() {
+        return Math.floor(Math.random() * 1000);
+    }
+    
+    getPortionNames() {
+        return this.portions.map((portion) => {
+            return portion.name;  
+        });
+    }
+    getPortionCosts() {
+        const ingredientsCost = this.portions.map((portion) => {
+            return (portion.amount * portion.cost).toFixed(2) / 100;
+        });
+        return ingredientsCost;
+    }
+    calcTotalRecipeCost() {
+        const storedIngredients = this.getPortionCosts();
+        const reducedIngredientCost = storedIngredients.reduce((acc, cur) => {
```

Comment:
could we rename acc to accumulator and cur to currentValue? 
alternatively, we could call them
sumOfAllPortionCosts and currentPortionCost

Reply:
Changed over, it is more empathetic now